### PR TITLE
Flow Based Computation API

### PR DESCRIPTION
**Pull request by pengbo-ruidian**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

Restructured and implemented Flow Based Computation API module

### CODE REVIEW

File: data/glsk-file/src/main/java/com/farao_community/farao/commons/data/glsk_file/actors/GlskDocumentLinearGlskConverter.java [hunk 1]
```diff
@@ -144,4 +145,11 @@
 
         return countryGlskDataChronologyMap;
     }
+
+    public Instant getInstantStart(String filePathString) throws IOException, ParserConfigurationException, SAXException {
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply data/glsk-file/src/main/java/com/farao_community/farao/commons/data/glsk_file/actors/GlskDocumentLinearGlskConverter.java]: removed

File: flowbased-computation/flowbased-computation-afs-local/src/main/java/com/farao_community/farao/flowbased_computation/afs/local/LocalFlowBasedComputationRunningService.java [hunk 2]
```diff
@@ -58,7 +58,10 @@ public void run(FlowBasedComputationRunner runner) {
             Network network = aCase.getNetwork();
 
             logger.log(RESOURCE_BUNDLE.getString("RunningFlowBased"));
-            FlowBasedComputation raoComputation = factorySupplier.get().create(network, cracFileProvider.getCracFile(), computationManager, 0);
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply flowbased-computation/flowbased-computation-afs-local/src/main/java/com/farao_community/farao/flowbased_computation/afs/local/LocalFlowBasedComputationRunningService.java]: reverted

File: flowbased-computation/flowbased-computation-api/pom.xml [hunk 3]
```diff
@@ -25,6 +25,14 @@
             <groupId>com.powsybl</groupId>
             <artifactId>powsybl-iidm-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-iidm-xml-converter</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-iidm-impl</artifactId>
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply flowbased-computation/flowbased-computation-api/pom.xml]: moved to scope runtime

File: data/glsk-file/src/test/java/com/farao_community/farao/commons/data/glsk_file/GlskDocumentLinearGlskConverterTest.java [hunk 4]
```diff
@@ -88,4 +89,10 @@ public void testConvertGlskDocumentToLinearGlskDataChronologyFromFilePath() thro
         Path pathtest = Paths.get("src/test/resources/GlskB42CountryIIDM.xml");
         assertTrue(!new GlskDocumentLinearGlskConverter().convertGlskDocumentToLinearGlskDataChronologyFromFilePath(pathtest, testNetwork).isEmpty());
     }
+
+    @Test
+    public void testGetStartPoint() throws ParserConfigurationException, SAXException, IOException {
+        Path pathtest = Paths.get("src/test/resources/GlskB42CountryIIDM.xml");
+        assertTrue((new GlskDocumentLinearGlskConverter().getInstantStart(pathtest.toString())).toString().equals("2018-08-28T22:00:00Z"));
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply data/glsk-file/src/test/java/com/farao_community/farao/commons/data/glsk_file/GlskDocumentLinearGlskConverterTest.java]: moved

File: flowbased-computation/flowbased-computation-api/pom.xml [hunk 5]
```diff
@@ -25,6 +25,14 @@
             <groupId>com.powsybl</groupId>
             <artifactId>powsybl-iidm-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-iidm-xml-converter</artifactId>
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply flowbased-computation/flowbased-computation-api/pom.xml]: changed to scope runtime

**Comment by baptsegu** [comment]
general comment : add/replace the authors with your name/address for all updated/created file

File: flowbased-computation/flowbased-computation-api/src/main/java/com/farao_community/farao/flowbased_computation/FlowBasedComputationFactory.java [hunk 6]
```diff
@@ -10,24 +10,41 @@
...
+    FlowBasedComputation create(Network network,
+                                CracFile cracFile,
+                                FlowBasedGlskValuesProvider flowBasedGlskValuesProvider,
+                                Instant instant,
+                                ComputationManager computationManager,
+                                int priority);
 
-    FlowBasedComputation create(Network network, CracFile cracFile, ComputationManager computationManager, int i);
+    /**
+     * @param network reference network: we need a network to construct the linear glsk map from the glsk document
```
[the remark this answers is not in the document]
Comment (pengbo-ruidian) [reply flowbased-computation/flowbased-computation-api/src/main/java/com/farao_community/farao/flowbased_computation/FlowBasedComputationFactory.java]: OK. refer to commit refactoring 17dc568.

**Pull request closed without merging**